### ISSUE TITLE
Integrate ETHOS Suite with ETHOS Simulator for Model Editing

### ISSUE THREAD

**Issue by frskyrc-ben** [FrSkyRC/ETHOS-Feedback-Community#4588]
One of the most requested features people have asked for is the ability to edit models ie. a "Companion" style app.

We may wish to considering linking ETHOS Suite software to the ETHOS Simulator software in a logical way to allow model editing.

For example, when the user connects a radio, ETHOS Suite can check the model of radio and suggest installing the accompanying Simulator version for that radio under the Model Manager area.

Once the matching Simulator version has been installed, ETHOS Suite should Sync the model files from the radio to the Simulator model file location. (#2911 / #3440)

When the user clicks "Edit" on a particular model, ETHOS Simulator should start up with the model pre-activated.

After making any changes to the model file via ETHOS Simulator and closing the simulator, ETHOS Suite should prompt the user if they want to sync the updated model file from the ETHOS Simulator back to the radio system

Here are a few simplified basic screenshot mockups for how I could envision this working.

![image](https://github.com/user-attachments/assets/d4b1d89c-64d0-4d03-82bd-82e71f97eb1d)

Of course, a fully-featured model editor will be preferred allowing for Copy/Paste between models and advanced model editing (which are features requested on other related GitHub issues, #3880 #1440 #1282)

As well, the controls for ETHOS Simulator should be improved to allow the Simulator to recognize the stick and switch actions from the radio directly or by having virtual mockup controls, instead of using the Keyboard layout which is currently the only way. (#4587 )

This is just an idea on how I could envision the two separate softwares being linked to provide greater functionality between them.

**Comment by pmullen503** [comment]
Agreed. Companion (with built in simulator) made OTX so much easier to use.

While I appreciate all the added features of Ethos, bringing Suite up to the level of companion and beyond will benefit the most users immediately.

**Comment by Richrfl** [comment]
This is a brilliant idea. We need this capability, the sooner, the better.

**Comment by rcgecko** [comment]
Excellent idea!!

**Comment by EngelMT** [comment]
I know that our customers wish such "Companion" too. So your idea is great and can be a first step.

**Comment by Richrfl** [comment]
Ben, when can we expect the implementation for his enhancement?

**Comment by frskyrc-ben** [comment]
> Ben, when can we expect the implementation for his enhancement?

I can't say for certain, but I will bring this issue to the attention of the dev working on ETHOS suite to see if we can provide a roadmap to a solution.

**Comment by Richrfl** [comment]
> > Ben, when can we expect the implementation for his enhancement?
> 
> I can't say for certain, but I will bring this issue to the attention of the dev working on ETHOS suite to see if we can provide a roadmap to a solution.

I appreciate your feedback, Ben. Integrating the Suite with the Sim is a good idea, although that does not seem to be an easy job. This kind of patch may be a nightmare for developers and users. The "SimSuite" package would be easier to handle from scratch while keeping the Suite and Sim alive and overlapping until the new SW is reasonably stable.

**Comment by Nicholas-Luoyi** [comment]
We haven't forgotten about this feature and have been discussing its implementation. We have also considered the method you suggested. However, we're currently facing some issues, such as

- Model file version compatibility
- The need to compile simulators for all types of radio on all platforms (Linux, Windows, macOS)
- The fact that every Ethos version update would require re-downloading the simulator

And other challenges. While we can't give you a definite timeline for its official release, we can certainly assure you that we also want this feature and are working on resolving these issues.

**Comment by Richrfl** [comment]
> We haven't forgotten about this feature and have been discussing its implementation. We have also considered the method you suggested. However, we're currently facing some issues, such as
> 
> * Model file version compatibility
> * The need to compile simulators for all types of radio on all platforms (Linux, Windows, macOS)
> * The fact that every Ethos version update would require re-downloading the simulator
> 
> And other challenges. While we can't give you a definite timeline for its official release, we can certainly assure you that we also want this feature and are working on resolving these issues.

Thank you very much for your response Nick. I do not want to be a smart A**, but rather to drop some ideas, therefore please do not get me wrong with my suggestions below:
About the model file version, is about the same as we have today, mostly with the Sim. The Suit seems to be pretty the same for several versions, therefore as I guess that the way you design is in "blocks" yes, there will be one part (the Sim side) that will require to be compiled with each FW version( like today). The real more complex part, as it will require large "branches" in the SW, will be to select the TX version, based on the Radio model, which will make the SW package pretty big, but as I said before overlapping the current SIM/Suite with the new one, will allow adding models one at a time if that is what it takes. 
Having to support several platforms is also a pain in the rear, but again, no need to have them all in the first release.
Again, my apologies if I am going over my head. Unfortunately, I am not very good at writing code; otherwise, I would happily offer my time to help you guys.

**Comment by JustPlaneChris** [comment]
I'd just like to add my vote to a "companion-like" addition to the Ethos Suite to allow easy model editing and copying logic switches and such from one model to another. I very much miss that ability from EdgeTX.

**Comment by fdm225** [comment]
Not to be a downer on this but since there the simulator doesn't support MACOS this would be truly disappointing.  I get that windows runs the world but we MAC users still buy your products too.  Please focus on an actual addition to ethos suite rather than giving windows additional features.

**Comment by Kriske46** [comment]
This is indeed a musthave! I’m trying to copy widget screens from one model to another. But this is not possible. I would expext this to be a basic feature. If there would be a model editor, this would make this so easy. The way todo this now is so frustrating and time consuming. Hopefully this will be added very soon.

**Comment by fdm225** [comment]
I agree, I just think they need to include macOS in their planning and since they can’t seem to get a simulator for Mac then this may not be the way.

**Comment by Devil-IdT** [comment]
This feature has been requested for so long. I'm surprised that no assignees, label, or type has been specified yet.
Has this topic perhaps already been accepted in another issue or set as a milestone?

Perhaps @bsongis  could comment on what the timeline looks like. ;)

Best regards, Micha

**Comment by bsongis** [comment]
This feature is in progress ...

https://github.com/user-attachments/assets/a0daf12c-29cb-48e4-9343-ef0d839190f6

**Comment by Kriske46** [comment]
Wow! This is so cool!!!! Looking forward to play with this editor.

**Comment by Devil-IdT** [comment]
> This feature is in progress ...
> 
>  model-editor.mp4

Hi Bertrand, that looks really good. Will there be a “Print Model” function again?

![Image](https://github.com/user-attachments/assets/4a44e36b-d5a9-4c3e-b36d-17572aa6331c)

Best regards, Micha

**Comment by bsongis** [comment]
Yes I think we can do better than Companion. We just need time to implement everything and in the same time bring more features.

**Comment by Devil-IdT** [comment]
Thank you very much, that sounds great.
I hope you have lots of success and wish I could help you with your fantastic work.

Best regards, Micha
